### ISSUE TITLE
RUF007 violates naming convention

### ISSUE THREAD

**Issue by dylwil3** [astral-sh/ruff#12397]
[pairwise-over-zipped (RUF007)](https://docs.astral.sh/ruff/rules/pairwise-over-zipped/) appears to violate the [naming convention](https://github.com/astral-sh/ruff/blob/main/CONTRIBUTING.md#rule-naming-convention), unless I'm misunderstanding either the rule or the convention.

Perhaps `zip-over-pairwise` or `zip-for-pairwise` would be more appropriate? Happy to make the change, or have the issue closed if this seems like not a big deal.

Here is the rule to save you a click:

> When iterating over successive pairs of elements, prefer `itertools.pairwise()` over `zip()`.
> 
> 
> ```python
> letters = "ABCD"
> zip(letters, letters[1:])  # ("A", "B"), ("B", "C"), ("C", "D")
> ```
> 
> Use instead:
> 
> ```python
> from itertools import pairwise
> 
> letters = "ABCD"
> pairwise(letters)  # ("A", "B"), ("B", "C"), ("C", "D")
> ```

And here is the relevant part of the naming convention:

> Like Clippy, Ruff's rule names should make grammatical and logical sense when read as "allow ${rule}" or "allow ${rule} items", 
> as in the context of suppression comments.
> 
> For example, AssertFalse fits this convention: it flags assert False statements, and so a suppression comment would be framed as "allow assert False".

**Comment by charliermarsh** [comment]
Yeah this seems wrong (and we can change it since the human-readable names aren't part of our versioning policy).

**Comment by charliermarsh** [comment]
I think `zip-instead-of-pairwise` would be most consistent with other rules.

**Comment by charliermarsh** [comment]
If you're interested in updating, I'd happily merge a PR.

**Comment by dylwil3** [comment]
Sure - on it!
